osu.Game/Screens/Edit/LegacyEditorBeatmapPatcher.cs

```diff
@@ -41,26 +41,28 @@ public void Patch(byte[] currentState, byte[] newState)
             foreach (var block in result.DiffBlocks)
             {
                 // Removed hitobjects
-                for (int i = 0; i < block.DeleteCountA; i++)
-                {
-                    int hoIndex = block.DeleteStartA + i - oldHitObjectsIndex - 1;
+                if (oldHitObjectsIndex != -1)
```

Comment:
Really should be `> 0`.

Also please add braces for this and the other statement. CI/code style analysis will complain about this 100%, and having a one-line condition followed by a multi-line loop is not readable.